File: CSharp/Student_Result_Processing_System/Models/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices.WindowsRuntime;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Models
{
    public class Student
    {
        public string name { get; private set; }
        public string rollNo { get; private set; }
        public List<int> marks { get; private set; }

        public Student(string name, string rollNo, List<int> marks)
        {
            this.name = name;
            this.rollNo = rollNo;
            this.marks = marks;
        }

        public int GetTotalMarks()
        {
            int total = 0;
            foreach(int m in marks)
            {
                total += m;
            }
            return total;
        }

        public double GetAverage()
        {
            return GetTotalMarks() / marks.Count;
        }

        public string getGrade()
        {
            int avg = (int)GetAverage();

            if (avg >= 90) return "A";
            else if (avg >= 75) return "B";
            else if (avg >= 60) return "C";
            else if (avg >= 35) return "D";
            else return "F";
        }


        public bool isPass()
        {
            //if (marks.Count == 0)
            //{
            //    Console.WriteLine("No marks are given.");
            //    return false;
            //}

            foreach(int m in marks)
            {
                if(m < 35)
                {
                    return false;
                }
            }

            return true;
        }

        public void DisplayInfo()
        {
            string status = isPass() ? "Pass" : "Fail";
            Console.WriteLine($"Roll No: {rollNo} | Name: {name} | Marks: [{string.Join(", ", marks)}]" + 
                $"Average: {GetAverage():F2} | Grade: {getGrade()} | Status: {status}");
        }

        public string GenerateReportCardText()
        {
            string status = isPass() ? "Pass" : "Fail";
            var lines = new List<string>();
            lines.Add("--- Report Card ---");
            lines.Add($"Name: {name}");
            lines.Add($"Roll No: {rollNo}");
            lines.Add($"Subject 1 Marks: {marks[0]}");
            lines.Add($"Subject 2 Marks: {marks[1]}");
            lines.Add($"Subject 3 Marks: {marks[2]}");
            lines.Add($"Total Marks: {GetTotalMarks()}");
            lines.Add($"Average: {GetAverage():F2}");
            lines.Add($"Grade: {getGrade()}");
            lines.Add($"Status: {status}");
            return string.Join("\n", lines);
        }
    }
}

File: Python/Student_Result_Processing_System/models/student.py
class Student:
    def __init__(self, name, roll_no, marks):
        self._name = name
        self._roll_no = roll_no
        self._marks = marks  # list of 3 numbers

    def is_pass(self):
        for mark in self._marks:
            if mark < 35:
                return False
        return True

    def get_total_marks(self):
        total = 0
        for mark in self._marks:
            total += mark
        return total

    def get_average(self):
        return self.get_total_marks() / len(self._marks)

    def get_grade(self):
        average = self.get_average()
        if average >= 90:
            return "A"
        elif average >= 75:
            return "B"
        elif average >= 60:
            return "C"
        elif average >= 35:
            return "D"
        else:
            return "F"

    def display_info(self):
        status = "Pass" if self.is_pass() else "Fail"
        print(f"Roll No: {self._roll_no} | Name: {self._name} | Marks: {self._marks} | "
              f"Total: {self.get_total_marks()} | Average: {self.get_average():.2f} | "
              f"Grade: {self.get_grade()} | Status: {status}")
        
    def generate_report_card_text(self):
        status = "Pass" if self.is_pass() else "Fail"
        lines = []
        lines.append("--- Report Card ---")
        lines.append(f"Name: {self._name}")
        lines.append(f"Roll No: {self._roll_no}")
        lines.append(f"Subject 1 Marks: {self._marks[0]}")
        lines.append(f"Subject 2 Marks: {self._marks[1]}")
        lines.append(f"Subject 3 Marks: {self._marks[2]}")
        lines.append(f"Total Marks: {self.get_total_marks()}")
        lines.append(f"Average: {self.get_average():.2f}")
        lines.append(f"Grade: {self.get_grade()}")
        lines.append(f"Status: {status}")
        return "\n".join(lines)